engine/state_manager.go

```diff
@@ -0,0 +1,136 @@
+package engine
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/state"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+const (
+	StateManagement          = "StateManagement"
+	DefaultStateManagerDelay = time.Minute
+)
+
+var (
+	errSubsystemNotSetup = errors.New("subsystem not set up")
+	errNilInterface      = errors.New("interface is nil")
+)
```

Comment:
The first error already exist as `ErrNilSubsystem` and the second one can be moved to `subsystem_types`